src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -100,6 +96,16 @@ public virtual void WriteCsvLine(string line)
         /// </summary>
         protected override void BeginProcessing()
         {
+            if (this.MyInvocation.BoundParameters.ContainsKey("IncludeTypeInformation") && this.MyInvocation.BoundParameters.ContainsKey("NoTypeInformation"))
```

Comment:
Interesting, can we replace the _string_ names with strong typed names?

What about different parameter sets?

Reply:
>  can we replace the string names with strong typed names?

I'm not sure I understand what you are asking. `IncludeTypeInformation.GetType().Name` instead of `"IncludeTypeInformation"`?

> What about different parameter sets?

parameter sets wont work without creating a bunch more. The parameter sets are currently being used for culture vs delimiter. It would get messy and harder to maintain than a logic check, IMO.

Reply:
In the future, we need to revisit how to make defining and reading mutual exclusion of parameters easier.  Using multiple parametersets today can be quite complex to get right.  For now, having a runtime check is ok.

Reply:
@SteveL-MSFT I created #5175 to track/discuss mutually exclusive parameters form a general position.

Reply:
I meant something like `nameof(param1)`.

Reply:
@iSazonov thanks. fixed.